File: LeetCode/Python/Easy/1165-Single-Row-Keyboard/solution.py
class Solution:
    def calculateTime(self, keyboard: str, word: str) -> int:
        result=0

        hashmap=[0 for _ in range(0, 26)]

        for (index, character) in enumerate(keyboard):
            ascii_index=ord(character) - ord('a')

            hashmap[ascii_index]=index

        previous_key_index=0

        for character in word:
            character_ascii=ord(character) - ord('a')

            key_index=hashmap[character_ascii]

            result+=abs(key_index - previous_key_index)

            previous_key_index=key_index

        return result

File: LeetCode/Rust/Easy/1165-Single-Row-Keyboard/solution.rs
impl Solution
{
    pub fn calculate_time(keyboard: String, word: String) -> i32
    {
        let mut result: i32=0;

        let mut hashmap: Vec<usize>=vec![0; 26];

        let offset: usize='a' as usize;

        for (index, character) in keyboard.chars().enumerate()
        {
            let ascii_index: usize=character as usize - offset;

            hashmap[ascii_index]=index;
        }

        let mut previous_key_index: usize=0;

        for character in word.chars()
        {
            let ascii: usize=character as usize - offset;

            let key_index: usize=hashmap[ascii];

            let difference: i32=(key_index as i32 - previous_key_index as i32).abs();

            result+=difference;

            previous_key_index=key_index;
        }

        result
    }
}